packages/smooth_app/lib/pages/product/simple_input_text_field.dart

```diff
@@ -79,6 +79,7 @@ class _SimpleInputTextFieldState extends State<SimpleInputTextField> {
       if (_manager == null || search.length < widget.minLengthForSuggestions) {
         _suggestions[search] = _SearchResults.empty();
       } else {
+        _setLoading(true);
```

Comment:
What about `_setLoading(false);` in the `catch`?

Reply:
I would say no, since there is a `_setLoading(false)` call, that will be always called after try-catch block.

But there is a bug here. If exception is thrown, `_suggestions[search]` will be null, and it will not disable loading and also crash the app on the last return. Will fix it.